### PR TITLE
Fix missing conversion of unmutated parts

### PR DESCRIPTION
**Pull request by amosbird**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Bug Fix

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix possible server crash after `ALTER TABLE ... MODIFY COLUMN ... NewType` when `SELECT` have `WHERE` expression on altering column and alter doesn't finished yet.

### CODE REVIEW

**Comment by amosbird** [comment]
Hmm,

```
docker: Error response from daemon: toomanyrequests: You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit.
```

**Review by alesapin — APPROVED** [review]
LGTM.

I'll backport it into all versions starting from 20.3.

**Comment by alesapin** [comment]
Nice flaky check catch.

File: src/Storages/MergeTree/MergeTreeRangeReader.cpp [hunk 1]
```diff
@@ -635,30 +635,34 @@ MergeTreeRangeReader::ReadResult MergeTreeRangeReader::read(size_t max_rows, Mar
...
-            auto block = prev_reader->sample_block.cloneWithColumns(read_result.columns);
-            auto block_before_prewhere = read_result.block_before_prewhere;
-            for (auto & ctn : block)
-            {
-                if (block_before_prewhere.has(ctn.name))
-                    block_before_prewhere.erase(ctn.name);
-            }
-
-            if (block_before_prewhere)
+            /// If some columns absent in part, than evaulate default values
```
Comment (amosbird) [inline src/Storages/MergeTree/MergeTreeRangeReader.cpp:640]: then )

**Pull request merged**